server/routes/getRewards.ts

```diff
@@ -46,39 +43,96 @@ const getRewards = async (models, req: Request, res: Response, next: NextFunctio
     startDate = startDate.toISOString(); // 2020-08-08T12:46:32.276Z FORMAT // 30 days ago date
   }
 
-  const rewards = await models.ChainEvent.findAll({
-    where: {
+
+  const chainInfo = await models.Chain.findOne({ where: { id: chain } });
+
+
+  // if we are using old version of chain
+  if (chain == 'edgeware' && version < 34){
+
+    // there is no validator identifier in reward event in chain-event for old version,
+    // get the sessions and rewards over specified time from chain-event and 
+    // join session event with rewards
+    // if a stash id is provided then filter out just the session having provided
+    // stash id
+    let rawQuery = `
+      SELECT *
+      FROM(
+        SELECT block_number  as "session_block_number", event_data as "session_event_data", row_number()
+        OVER (ORDER BY created_at DESC) AS "rn"
+        FROM "ChainEvents" ce
+        WHERE chain_event_type_id  = '${chain}-new-session' AND created_at BETWEEN '${startDate}' AND '${endDate}'
+      ) newSession
+      JOIN (
+        SELECT block_number  as "reward_block_number", event_data as "reward_event_data", row_number()
+        OVER (ORDER BY created_at DESC) AS "rn"
+        FROM "ChainEvents" ce
+        WHERE chain_event_type_id  = '${chain}-reward' AND created_at BETWEEN '${startDate}' AND '${endDate}'
+      ) reward
+        ON reward.rn = newSession.rn`
+      
+      if (stash_id){
+        rawQuery += ` where newSession.session_event_data -> 'data' ->> 'active' LIKE '%${stash_id}%'`;
+      }
+
+      session_rewards = await sequelize.query(rawQuery, {});
+      session_rewards = session_rewards[0] //fetch the results
+      
+      // if no session/reward found 
+      if (session_rewards.length == 0) { return res.json({ status: 'Success', result: { validators: validators || [] } }); }
+
+
+      session_rewards.forEach((s_r) => {
+        const reward_event_data: IEventData = s_r.reward_event_data;
+        const block_number = s_r.reward_block_number;
+        const session_validators = s_r.session_event_data.data.active ;
+        session_validators.forEach((validator_id) => {
+          if (!Object.prototype.hasOwnProperty.call(validators, validator_id)){ 
+            validators[validator_id] = {}; 
+          }
+          validators[validator_id][block_number] = reward_event_data['amount'];
+        });
+      });
+
+      if(stash_id){
+        validators = {stash_id:validators[stash_id]}
```

Comment:
spacing should be `validators = { stash_id: validators[stash_id] }

Reply:
rectified.